File: backend/app/schemas/news.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

NewsSource = Literal["xuangubao", "jiuyangongshe", "xueqiu"]


class NewsItemRead(BaseModel):
    id: str
    source: NewsSource
    title: str
    summary: str = ""
    published_at: datetime | None = None
    author: str = ""
    url: str = ""
    symbol_keyword: str = ""

    model_config = ConfigDict(from_attributes=True)


class NewsFeedResponse(BaseModel):
    items: list[NewsItemRead]
    errors: list[str] = []


class NewsBriefResponse(BaseModel):
    market: list[NewsItemRead]
    discussions: list[NewsItemRead]
    xueqiu: list[NewsItemRead]
    errors: list[str] = []

File: frontend/src/types/news.ts
export type NewsSource = 'xuangubao' | 'jiuyangongshe' | 'xueqiu'

export interface NewsItem {
  id: string
  source: NewsSource
  title: string
  summary: string
  published_at: string | null
  author: string
  url: string
  symbol_keyword: string
}

export interface NewsFeedResponse {
  items: NewsItem[]
  errors: string[]
}

export interface NewsBriefResponse {
  market: NewsItem[]
  discussions: NewsItem[]
  xueqiu: NewsItem[]
  errors: string[]
}
